### DISCUSSION TITLE
How can I add Japanese language ?

### DISCUSSION THREAD

**Discussion by mynulislam2** [mindee/doctr#1468] [Q&A]
**Basically I am trying to add Japanese font with English number and English characters** 

```
#!/usr/bin/env python
# -*- coding: utf-8 -*- 
from doctr.io import DocumentFile
from doctr.models import ocr_predictor,from_hub
import json
from doctr.datasets import vocabs



model = ocr_predictor(det_arch = 'db_resnet50',    
                      reco_arch = 'crnn_vgg16_bn', 
                      pretrained=True
                     )

img_path = "auctionsheet.jpg" #Specify your image path here
img = DocumentFile.from_images(img_path)
print(img)
result = model(img)
output = result.export()

with open("path/docTR_OCR_output.json", "w") as f:
    f.write(json.dumps(output, indent=1))
f.close()

result.show(img)
```
  **this are my codes** 

**I think i  might need to do some thing like this** 
```

japanesevocabs=  vocabs.VOCABS["japanese"] = (
    vocabs.VOCABS["digits"]
    + vocabs.VOCABS["ascii_letters"]
    + "あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをん"
    + "アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホマミムメモヤユヨラリルレロワヲン"
    + "、。！？ー"
)

```
**but how can i use it ?** 
```
model = ocr_predictor(det_arch = 'db_resnet50',    
                      reco_arch = 'crnn_vgg16_bn', 
                      pretrained=True,
                      vocabs=japanesevocabs
                     )  
```
I tried it like this but it's gives error 

also how can I train my model for this language?

**Comment by felixdittrich92** [comment]
Hi @mynulislam2 :wave:,

All models are pretrained on french vocab so you would need to train your own model / models
The detection works well in your case so we are talking only about the recognition ?

Best regards,
Felix

training:
https://mindee.github.io/doctr/using_doctr/custom_models_training.html

**Reply by mynulislam2** [reply]
![Screenshot (40)](https://github.com/mindee/doctr/assets/68342123/197e1a79-7a41-4e60-975d-30b413883f33)
This is how it's working . you can see it can detect the texts but when I am hovering any text it shows in English character instead of Japanese text . so if trains custom models it will work? and more question is that is there any way that i will bring external model with doctr?

**Reply by felixdittrich92** [reply]
If you have some labeled datasets for detection (to improve the current results) and for recognition the models should be able to learn it (for recognition i suggest to use `parseq` instead of a `crnn` model if you start to train your own one)

Keep in mind handwritten texts are always a bit tricky (different spaces between characters, a lot of different fonts, etc.)

What do you mean with external model ? :)

**Reply by O18D** [reply]
Can anyone of you provide me japanese invoice dataset
